python-package/lightgbm/basic.py

```diff
@@ -1174,6 +1175,28 @@ def num_feature(self):
         else:
             raise LightGBMError("Cannot get num_feature before construct dataset")
 
+    def get_ref_chain(self, ref_limit=100):
+        '''
+        Gets a chain of Dataset objects, starting with r, then going to r.reference if exists,
+            then to r.reference.reference, etc. until we hit ref_limit or a reference loop
+
+        Returns
+        -------
+        chain of references of self : set of Dataset objects
+        '''
+        head = self
+        ref_chain = []
```

Comment:
@j-mark-hou you can open another one. Can you also add a simple test case for your issue in test folder (or modify an existed one maybe)?